Instagram/View Controllers/HomeViewController.m

```diff
@@ -74,14 +80,21 @@ - (void) queryPosts {
     }];
 }
 
-/*
+- (void)refreshData:(UIRefreshControl *)refreshControl {
+    [self queryPosts];
+    [refreshControl endRefreshing];
```

Comment:
This seems off. Have you tested with bad network? Think about a timeline:

```
| Refresh action taken - control starts spinning.
| Post query starts
| Refresh control ends refreshing (stops spinning and hides)
|
| User sees no spinning control and also no new posts because of network latency 
|
| Post query ends